### DISCUSSION TITLE
Adonisjs  Rest API and React Client

### DISCUSSION THREAD

**Discussion by feniljariwala82** [adonisjs/core#2882] [Help] [answered]
Currently, I have been working on a project which is based on the MERN stack. But due to the strong database configuration of adonis, I want to switch to the Adonisjs Rest API with React as a client library. Does anyone have any experience in this? My main purpose is to run Adonisjs API and React as a web app. Your guidance will be highly appriciated

**Comment by thetutlage** [comment, marked as answer]
Not sure how you have been structuring your projects so far. But I usually go with two different apps. One is the API server (AdonisJS) and the other one is the frontend codebase, which is react in your case.

Your Adonis server just has to return JSON and React has to consume. I don't see any particular problems or challenges you will face there

**Reply by feniljariwala82** [reply]
I have always used React as a client library with Express.js, where I used to host React + Express.js on one server. And, I didn't know how Adonisjs and React(Static assets) can host on the same domain. I got your point. Thank you for your guidance.

**Reply by RomainLanz** [reply]
You can also use something like [InertiaJS](https://inertiajs.com) to have everything on the same server.
